src/client/Client.java

```diff
@@ -225,9 +233,46 @@ private void updateButtons() {
         interactionPanel.repaint();
     }
 
+    private void updateTimer(boolean endOfText) {
+        if (currScene.timer != null) {
```

Comment:
Need to reset the counter on every new scene

Reply:
Good catch